### PR TITLE
ci: replace isort with ruff

### PR DESCRIPTION
**Pull request by epenet**
## Checklist

- [ ] This PR has sufficient documentation.
- [ ] This PR has sufficient test coverage.
- [x] This PR title satisfies semantic [convention](https://www.conventionalcommits.org/en/v1.0.0/#summary).

## Additional Comments

No additional comments.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -104,6 +89,16 @@ source = ['./src']
 [tool.coverage.report]
 exclude_lines = ['pragma: no-cover', 'if TYPE_CHECKING:', '@abstractmethod']
 
+[tool.ruff]
+# Set the maximum line length to 88.
+line-length = 88
+
+[tool.ruff.lint]
+# Set the maximum line length to 88.
```
Comment (epenet) [inline pyproject.toml:97]:
```suggestion
```

File: pyproject.toml [hunk 2]
```diff
@@ -104,6 +89,15 @@ source = ['./src']
 [tool.coverage.report]
 exclude_lines = ['pragma: no-cover', 'if TYPE_CHECKING:', '@abstractmethod']
 
+[tool.ruff]
+# Set the maximum line length to 88.
+line-length = 88
+
```
Comment (epenet) [inline pyproject.toml:95]:
```suggestion
```

**Comment by epenet** [comment]
@noahnu what do you think of this isort migration?
One step towards moving more to ruff...

**Review by noahnu — APPROVED** [review]

**Pull request merged**